packages/lesswrong/components/common/withDialog.tsx

```diff
@@ -4,12 +4,10 @@ import { hookToHoc } from '../../lib/hocUtils';
 import { useTracking } from '../../lib/analyticsEvents';
 import { useOnNavigate } from '../hooks/useOnNavigate';
 
-export type CloseableComponents = {
-  [T in keyof ComponentTypes]: FromPartial<ComponentTypes[T]['propTypes']> extends { onClose: any } | undefined ? T : never
-}[keyof ComponentTypes];
+export type CloseableComponent = ComponentWithProps<{ onClose?: any }>
```

Comment:
I refactored this type definition into `ComponentWithProps`, which I also used to enforce the type of form group layout components. `ComponentWithProps` no longer uses the legacy [propTypes](https://legacy.reactjs.org/docs/typechecking-with-proptypes.html) property to check the type